tests/performance/cli.py

```diff
@@ -0,0 +1,120 @@
+from datetime import datetime, timedelta
+import enum
+import os
+from pathlib import Path
+from typing import Optional
+
+from dotenv import find_dotenv, load_dotenv
+from rich.console import Console
+from rich.table import Table
+from tqdm import tqdm
+import typer
+from loguru import logger
+
+from cloudpathlib import implementation_registry
+
+
+from runner import main, normalize_results, results_to_csv
+
+# make loguru and tqdm play nicely together
+logger.remove()
+logger.add(lambda msg: tqdm.write(msg, end=""), colorize=True)
+
+# get environement variables
+load_dotenv(find_dotenv())
```

Comment:
I personally dislike automatic `.env` loading like this, and prefer having it be more explicit / closer to the user. For example, have the make command use the dotenv CLI with `dotenv -f .env run python tests/performance/cli.py s3 --save-csv=perf-results.csv`. Not a position I hold strongly.

Reply:
Yeah, agreed this should be as close to the user as possible (which is why we have it just at the `cli.py` entrypoint and `conftest`). The library code does not explicitly do anything with `.env`. I have a slight preference for loading this at the entrypoint since I don't always use these commands through the makefile, and I expect my `.env` to get loaded (and I also want parsimonious cli commands).